### PR TITLE
update README.md

### PR DESCRIPTION
**Pull request by viv-defy**

### CODE REVIEW

**Review by horsey-defy — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -21,46 +46,68 @@ If you are running this project on localhost, you need to have a local node runn
...
-1. navigate to `.env.example` and set the environment variables mentioned there (using the export command) (RUN_ENV can be one of "Development", "Production", "Staging")
-2. if your RUN_ENV is "Development", set INFURA_KEY to an empty string. You will also need to create a copy of config/Staging.toml and rename it to config/Development.toml. Set the values in the config file as per your requirements.
-3. setup a postgres database and update environment variable as per .env.example accordingly
-4. run `cargo run`
 
-<ins>NOTE</ins>: If there are any changes in the schema or the queries, run `cargo sqlx prepare --database-url $DATABASE_URL` and add the generated files or the github workflow will fail. Files will be generated under `bundler/.sqlx`  
+### Running the Relay/Bundler
+1. Set up a postgres database
+2. Navigate to `bundler/.env.example` and set the environment variables mentioned there (using the export command)
+   1. By default, the server uses "Staging.toml" as the config file. If you want to use a different config file, set the `RUN_ENV` environment variable to the path of the config file. `RUN_ENV` can be one of:
```
Comment (ska1296) [inline README.md:67]: we use Development.toml be default

File: README.md [hunk 2]
```diff
@@ -21,46 +46,68 @@ If you are running this project on localhost, you need to have a local node runn
...
+      1. Development
+      2. Production
+      3. Staging
+   2. If your RUN_ENV is "Development", set INFURA_KEY to an empty string
+   3. Use `key` obtained in the previous section to populate `WALLET_PRIVATE_KEY` and `VERIFYING_PAYMASTER_PRIVATE_KEY`
+3. Create a copy of `config/Staging.toml` and rename it as `config/Development.toml`. Set the values in the config file as per your requirements.
+   ```
+   cp config/Staging.toml config/Development.toml
+   ```
+4. Run
```
Comment (ska1296) [inline README.md:77]: also need to run migrations before this

File: README.md [hunk 3]
```diff
@@ -21,46 +46,68 @@ If you are running this project on localhost, you need to have a local node runn
...
 
 The project does not come with a "Production.toml", but you can create one and use it. The config file should be in the same format as "Development.toml".
 
 ## Account Abstraction Deployment & Testing
 
-A foundry project for deployment and testing of the Account Abstraction contracts
+A Foundry project for deployment and testing of the ERC-4337 contracts
 
 ### Set Up
+- git clone https://github.com/Club-Defy/roverx-rpc
```
Comment (ska1296) [inline README.md:95]: 1. this needs to be the very first step. It can then be removed from everywhere else.
2. Repo name is wrong

**Review by ska1296 — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -1,17 +1,42 @@
...
-
-### Running the node locally
-If you are running this project on localhost, you need to have a local node running with the contracts deployed. Here are the steps to set up your local node with the required contracts:
-1. follow the instructions in the [foundry installation guide](https://book.getfoundry.sh/getting-started/installation) to set up foundry tool kit
-2. navigate to the `contracts/` folder
-3. run `bash foundry_setup.sh` to install all the contract dependencies
-4. populate the `contracts/.env` with the following values
+Bundler is a component being built as per the [ERC4337](https://eips.ethereum.org/EIPS/eip-4337#rpc-methods-eth-namespace) (Account Abstraction via Entry Point Contract specification). Bundler is a Rust based implementation that exposes REST APIs and also acts as a Relayer.
+It uses Actix web framework to expose REST APIs.
+MSRP: `rustc 1.71.0 (8ede3aae2 2023-07-12)`
```
Comment (ska1296) [inline README.md:15]: would be better to make a pre requisite section and put this there, alongside foundry and postgres

File: README.md [hunk 5]
```diff
@@ -21,46 +46,68 @@ If you are running this project on localhost, you need to have a local node runn
...
+   The `PRIVATE_KEY` is the default key generated by anvil.
    The salt values can be changed if required. This will affect the addresses at which the contracts are deployed.
-5. run `bash script/deploy_local.sh`. This should start a local anvil node and deploy the contracts
+5. To start a local anvil node and deploy the contracts, run
+    ```
+    bash script/deploy_local.sh
+   ```
 
-You can find the deployed contracts in the console logs. Copy the very first private-key from the logs and use it to populate the values of `WALLET_PRIVATE_KEY` and `VERIFYING_PAYMASTER_PRIVATE_KEY` in env variables. This key is the deployer for all the contracts.
+On successful deployment, you can find the deployed contract addresses, `signer/owner` address and a `key`. Use this info to populate the `bundler/.env` and `bundler/config/*.toml` files, as directed in the following section
```
Comment (ska1296) [inline README.md:56]:
```suggestion
On successful deployment, you can find the deployed contract addresses, `signer/owner` address and a `key`. Use this information to populate the `bundler/.env` and `bundler/config/*.toml` files, as directed in the following section
```

better to be formal, I believe?

File: README.md [hunk 6]
```diff
@@ -21,46 +46,68 @@ If you are running this project on localhost, you need to have a local node runn
...
 
-### Running the bundler
-1. navigate to `.env.example` and set the environment variables mentioned there (using the export command) (RUN_ENV can be one of "Development", "Production", "Staging")
-2. if your RUN_ENV is "Development", set INFURA_KEY to an empty string. You will also need to create a copy of config/Staging.toml and rename it to config/Development.toml. Set the values in the config file as per your requirements.
-3. setup a postgres database and update environment variable as per .env.example accordingly
-4. run `cargo run`
 
-<ins>NOTE</ins>: If there are any changes in the schema or the queries, run `cargo sqlx prepare --database-url $DATABASE_URL` and add the generated files or the github workflow will fail. Files will be generated under `bundler/.sqlx`  
+### Running the Relay/Bundler
+1. Set up a postgres database
```
Comment (ska1296) [inline README.md:65]: Is it worth mentioning that we use version 13?

**Review by ska1296 — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (ska1296) [inline README.md:15]: Shouldn't this be "MSV"?

**Review by ska1296 — CHANGES_REQUESTED** [review]

**Comment by ska1296** [comment]
already merged to main in #401 and #403

**Comment by ska1296** [comment]
already merged to main in https://github.com/roverxio/toadplatform/pull/401 and https://github.com/roverxio/toadplatform/pull/403

**Pull request closed without merging**